wgpu-hal/src/dx12/command.rs

```diff
@@ -552,7 +552,24 @@ impl crate::CommandEncoder<super::Api> for super::CommandEncoder {
         for r in regions {
             let src_box = make_box(&wgt::Origin3d::ZERO, &r.size);
             unsafe {
-                *src_location.u.PlacedFootprint_mut() = r.to_subresource_footprint(dst.format)
+                let mut desc = dst.resource.GetDesc();
+                desc.Width = r.size.width as u64;
+                desc.Height = r.size.height;
+                desc.DepthOrArraySize = r.size.depth as u16;
+                let mut fp = d3d12_ty::D3D12_PLACED_SUBRESOURCE_FOOTPRINT::default();
+
+                self.device.GetCopyableFootprints(
```

Comment:
Instead of calling this fn we have `to_subresource_footprint` which is also used by `copy_texture_to_buffer`. I think we should instead see what's wrong with the calculations in `to_subresource_footprint` or use `GetCopyableFootprints` there.

Reply:
good point, I can't think of a reason as to not use `GetCopyableFootprints `, I'll refactor `to_subresource_footprint ` so it can use it.